data/catalogs/aws_data.yml

```diff
@@ -1,6 +1,8 @@
 ---
 meta:
-  version: 2023.2
+  version: 2024.1
+  hydromt_version: '>=0.9'
```

Comment:
So I'm not sure what version spec we want here. because if we release a new version of the catalog then this will still try to read that, but I'm also not sure how we should bound this version from above. We could add a `<0.10` for example, but it's not necessarifly true that 0.10 whill have braking changes for the catalog. @hboisgon what do you think?